runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c

```diff
@@ -160,29 +160,31 @@ static void genNumLocalesOptions(FILE* slurmFile, sbatchVersion sbatch,
 
 static void propagate_environment(FILE *f)
 {
+  // Indiscriminately propagate all environment variables.
+  // We could do this more selectively, but we would be likely
+  // to leave out something important.
+  char *enviro_keys = chpl_get_enviro_keys(',');
+  if (enviro_keys)
+    fprintf(f, " -E %s", enviro_keys);
+
+  // If any of the relevant character set environment variables
+  // are set, replicate the state of all of them.  This needs to
+  // be done separately from the -E mechanism because the launcher
+  // is written in Perl, which mangles the character set
+  // environment.
+  //
+  // Note that if we are setting these variables, and one or more
+  // of them is empty, we must set it with explicitly empty
+  // contents (e.g. LC_ALL= instead of -u LC_ALL) so that the
+  // launcher will not overwrite it.
```

Comment:
Somewhat nit-picking on phrasing ... strictly speaking it's not the launcher that will overwrite a variable in this case, but the _real itself.  However, that fact that it does so is (or would be) caused by the launcher's addition of a '-E' option.  To avoid a giant detour into philosophical notions of agency, maybe we could say "the Chapel launch mechanisms" rather than "the launcher"?

Reply:
Actually, it is the launcher that overwrites the variable, because the launcher is written in Perl.  The launcher then passes the overwritten version to the _real.

I like your solution.  I'll use that wording.